File: test/samples/keywords/keywords.rb
# +if_+:: not a condition
# +class_+:: not a class
# +i+:: just a string
# +in_+:: not in
# +for_+:: not a loop
# +words+:: contains lots of things
# +words_1+:: an integer
def keywords(if_, class_, i, in_, for_, words, words_1)
    # TODO If this compiles, it is already a good step!
end

if_ = STDIN.gets.to_i
class_ = STDIN.gets[0]
i = STDIN.gets.chomp
in_ = Hash[["a", "static"].zip(STDIN.gets.split.map(&:to_i))]
for_ = STDIN.gets.split.map(&:to_i)
words = Array.new(2) { {
    "int" => {
        "return" => STDIN.gets.to_i,
        "void" => STDIN.gets.split.map(&:to_i)
    },
    "if true" => STDIN.gets.to_i
} }
words_1 = STDIN.gets.to_i

keywords(if_, class_, i, in_, for_, words, words_1)

File: test/samples/keywords/keywords.pl
#!/usr/bin/perl
use strict;
use warnings;

# $if: not a condition
# $class: not a class
# $i: just a string
# %in: not in
# @for: not a loop
# @words: contains lots of things
# $words_1: an integer
sub keywords {
    my ($if, $class, $i, $in, $for, $words, $words_1) = @_;
    # TODO If this compiles, it is already a good step!
}

my $if = int <>;
my $class = substr <>, 0, 1;
my $i = <>;
chomp $i;
my @words2 = split /[ \n]/, <>;
my %in = ("a" => int($words2[0]), "static" => int($words2[1]));
my @for = map { int } split(/[ \n]/, <>);
my @words = ();
for (1..2) {
    push(@words, {});
    $words[-1]{'int'} = {};
    $words[-1]{'int'}{'return'} = int <>;
    $words[-1]{'int'}{'void'} = \@{[map { int } split(/[ \n]/, <>)]};
    $words[-1]{'if true'} = int <>;
}
my $words_1 = int <>;

keywords($if, $class, $i, \%in, \@for, \@words, $words_1);
